### PR TITLE
Fix numberWithCommas to handle precision properly

### PR DESCRIPTION
**Pull request by crabsinger**
Fix the numberWithCommas util to handle some edge cases. Add tests.

For example before this numberWithCommas(6, 2) returns 6, not 6.00.

### CODE REVIEW

**Review by levity — CHANGES_REQUESTED** [review]
there's already a `utils/index.test.ts`; please relocate the tests there. 

note about test style: i personally think it's ok to have a single test or a few tests with a number of assertions in them, especially when the assertions are self-explanatory and/or test names would be redundant with test content.

**Review by topocount — DISMISSED** [review]
thanks, this is great!

File: src/utils/index.test.ts [hunk 1]
```diff
@@ -1,8 +1,39 @@
 import { numberWithCommas } from './';
 
-test('numberWithCommas', () => {
```
Comment (crabsinger) [inline src/utils/index.test.ts:3]: @levity @topocount how's this for test styling?

Comment (crabsinger) [reply src/utils/index.test.ts:3]: happy for any nitty feedback on making it even better :)

**Review by zashton — APPROVED** [review]
